### PR TITLE
Ignore databricks_rag_studio in package mismatch

### PR DESCRIPTION
**Pull request by serena-ruan**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/serena-ruan/mlflow/pull/12231?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12231/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12231
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?
Ignore databricks_rag_studio package to avoid emitting warning

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/utils/requirements_utils.py [hunk 1]
```diff
@@ -627,13 +627,18 @@ def warn_dependency_requirement_mismatches(model_requirements: List[str]):
     doesn't satisfy them.
     """
     _DATABRICKS_FEATURE_LOOKUP = "databricks-feature-lookup"
+    _DATABRICKS_RAG_STUDIO = "databricks_rag_studio"
```
Comment (dbczumar) [inline mlflow/utils/requirements_utils.py:630]: @alkispoly-db @jameswu-db Can you confirm that this is the correct package name? Are there any other Databricks packages we should exclude from version mismatch warnings?

**Review by dbczumar — APPROVED** [review]
LGTM once we get confirmation from @alkispoly-db and @jameswu-db on https://github.com/mlflow/mlflow/pull/12231/files#r1625456165. Thanks so much, @serena-ruan !

File: mlflow/utils/requirements_utils.py [hunk 2]
```diff
@@ -627,13 +627,18 @@ def warn_dependency_requirement_mismatches(model_requirements: List[str]):
...
         for req in model_requirements:
             mismatch_info = _check_requirement_satisfied(req)
             if mismatch_info is not None:
                 # Suppress databricks-feature-lookup warning for feature store cases
-                if mismatch_info.package_name == _DATABRICKS_FEATURE_LOOKUP:
+                # Suppress databricks_rag_studio warning for RAG Studio cases
+                if mismatch_info.package_name in [
+                    _DATABRICKS_FEATURE_LOOKUP,
+                    _DATABRICKS_RAG_STUDIO,
+                ]:
```
Comment (harupy) [inline mlflow/utils/requirements_utils.py:641]: We might want to canonicalize this name. For pip, `foo-bar` is equivalent to `foo_bar`, for python, `"foo-bar"` is not `"foo_bar"`.

Comment (serena-ruan) [reply mlflow/utils/requirements_utils.py:641]: Do we infer the package name as 'foo-bar' if it is 'foo_bar'?

Comment (harupy) [reply mlflow/utils/requirements_utils.py:641]: Confirmed `mismatch_info.package_name` is not normalized.

Comment (harupy) [reply mlflow/utils/requirements_utils.py:641]: > Do we infer the package name as 'foo-bar' if it is 'foo_bar'?

Yes

**Review by WeichenXu123 — APPROVED** [review]
LGTM after Haru comment is addressed

File: tests/utils/test_requirements_utils.py [hunk 3]
```diff
@@ -555,7 +555,11 @@ def mock_get_installed_version_fn(package, module=None):
         mock_warning.assert_not_called()
 
 
-def test_suppress_warn_dependency_requirement_mismatches_feature_store(tmp_path):
+@pytest.mark.parametrize(
+    "ignore_package_name",
+    ["databricks-feature-lookup", "databricks_rag_studio", "databricks-rag-studio"],
```
Comment (harupy) [inline tests/utils/test_requirements_utils.py:560]:
```suggestion
    ["databricks-feature-lookup", "databricks_rag_studio", "databricks-rag-studio", "databricks.rag.studio", "Databricks-rag-studio"],
```

**Review by harupy — APPROVED** [review]
LGTM once @alkispoly-db @jameswu-db confirm this PR really fixes the issue.

File: mlflow/utils/requirements_utils.py (hunk 1 above)
Comment (alkispoly-db) [reply mlflow/utils/requirements_utils.py:632]: The package will be renamed. @mparkhe is the final name `databricks_rag_chains`?

Comment (mparkhe) [reply mlflow/utils/requirements_utils.py:632]: PyPI library: `databricks-chains`
Python package: `databricks.chains`

`databricks-rag-studio` and `databricks.rag_studio` will EoL

Comment (dbczumar) [reply mlflow/utils/requirements_utils.py:632]: ^ @serena-ruan can we make these adjustments?

Comment (serena-ruan) [reply mlflow/utils/requirements_utils.py]: Added `databricks-chains` and keep `databricks-rag-studio` for now, we can remove it after it's EoL

**Pull request merged**